src/main/java/magpiebridge/projectservice/java/InferSourcePath.java

```diff
@@ -104,18 +104,16 @@ public void accept(Path java) {
           return;
         }
 
-        if (!alreadyKnown(java)) {
```

Comment:
@ranjithmasthikatte  i didn't mean this one . But  changing  if (java.startsWith(root) && sourceRoots.get(root) > certaintyThreshold)  to  if (java.startsWith(root))

Reply:
@linghuiluo, as per our discussion, I updated the code and added the test case for sourceClassFullQualifiedNames counts in JavaProjectServiceTest. Only for MavenProjects added this test cases, since Gradle test case is ignored.